### PR TITLE
[BD-6170] Backpack price component improvement

### PR DESCRIPTION
**Pull request by colorwin**
Ticket: https://gojira.skyscanner.net/browse/BD-6170
Figma: https://www.figma.com/file/yN0hFyZlKL0Jwbpi0rEKYT/Backpack-Beta?node-id=4073%3A27622

Context: We should wrap the long price if it overflows the parent container, and use another prop to control the wrapping behavior for flexibility. 

Change description:
- The prop `size` only controls font size now
- Adding a new prop `align` to control the alignment and the wrapping behavior of `subtitle`
- “/ night” change to “a night” for the example
- Breaking the price if it is too long

Remember to include the following changes:

- [x] `UNRELEASED.md` (See [How to write a good changelog entry](https://github.com/Skyscanner/backpack/blob/main/CHANGELOG_FORMAT.md))
- [x] `README.md`
- [x] Tests
- [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

### CODE REVIEW

File: packages/bpk-component-price/README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default () => (
     size={SIZES.large}
     subtitle="£209"
     title="£1,830"
-    description="/ night"
+    description="a night"
```
Comment (Supremeyh) [inline packages/bpk-component-price/README.md:22]: should update `align` in `Props` of `README.md`

File: packages/bpk-component-price/src/BpkPrice.js [hunk 2]
```diff
@@ -23,11 +23,7 @@ import { cssModules } from 'bpk-react-utils';
 import BpkText, { TEXT_STYLES } from 'bpk-component-text';
 
 import STYLES from './BpkPrice.module.scss';
-
-export const SIZES = {
-  small: 'small',
-  large: 'large',
-};
+import { SIZES, ALIGNS } from './common-types';
```
Comment (Supremeyh) [inline packages/bpk-component-price/src/BpkPrice.js:26]: `align` is missing in ts `BpkPrice`  type `Props`

File: examples/bpk-component-price/examples.js [hunk 3]
```diff
@@ -18,55 +18,87 @@
...
+  SmallLeftExample,
+  SmallLeftWithDescriptionExample,
+  SmallLeftWithSubtitleDescriptionExample,
+  SmallRightExample,
+  SmallRightWithDescriptionExample,
+  SmallRightWithSubtitleDescriptionExample,
+  LargeLeftExample,
+  LargeLeftWithDescriptionExample,
+  LargeLeftWithSubtitleDescriptionExample,
+  LargeLeftLongPriceExample,
```
Comment (jasonhuang-sky) [inline examples/bpk-component-price/examples.js:102]: Please also add these two more cases
image

File: examples/bpk-component-price/examples.js [hunk 4]
```diff
@@ -18,55 +18,87 @@
...
     subtitle="£209"
     title="£1,830"
-    description="/ night"
+    description="a night"
   />
 );
 
+const LargeLeftLongPriceExample = () => (
+  // eslint-disable-next-line backpack/use-tokens
+  <div style={{ width: 170 }}>
```
Comment (jasonhuang-sky) [inline examples/bpk-component-price/examples.js:67]: Would be better to update to `184` to benefit the design to check it

File: UNRELEASED.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+**Breaking:**
+
+- bpk-component-price
+  - The prop `size` only control font size now
```
Comment (jasonhuang-sky) [inline UNRELEASED.md:4]:
```suggestion
  - The prop `size` only controls font size now
```

**Review by jasonhuang-sky — CHANGES_REQUESTED** [review]

**Review by jasonhuang-sky — APPROVED** [review]

File: packages/bpk-component-price/src/BpkPrice.js [hunk 6]
```diff
@@ -40,15 +37,17 @@ type Props = {
 const getClassName = cssModules(STYLES);
 
 const BpkPrice = (props: Props) => {
-  const { className, description, size, subtitle, title, ...rest } = props;
+  const { align, className, description, size, subtitle, title, ...rest } =
+    props;
 
   const isSmall = size === SIZES.small;
+  const isAlighRight = align === ALIGNS.right;
```
Comment (olliecurtis) [inline packages/bpk-component-price/src/BpkPrice.js:44]: Minor spelling mistake
```suggestion
  const isAlignRight = align === ALIGNS.right;
```

File: packages/bpk-component-price/src/BpkPrice.js [hunk 7]
```diff
@@ -40,15 +37,17 @@ type Props = {
...
 
   return (
     <div
       className={getClassName(
         'bpk-price',
-        isSmall && 'bpk-price--small',
+        isAlighRight && 'bpk-price--right',
```
Comment (olliecurtis) [inline packages/bpk-component-price/src/BpkPrice.js:50]:
```suggestion
        isAlignRight && 'bpk-price--right',
```

File: packages/bpk-component-price/src/BpkPrice.js [hunk 8]
```diff
@@ -63,9 +62,15 @@ const BpkPrice = (props: Props) => {
           {subtitle}
         </BpkText>
       )}
-      <div className={isSmall && getClassName('bpk-price__titleColumn')}>
+      <div
+        className={isAlighRight && getClassName('bpk-price__columnContainer')}
```
Comment (olliecurtis) [inline packages/bpk-component-price/src/BpkPrice.js:66]: Fixing spelling and SCSS suggestion made here: 
```suggestion
        className={isAlignRight && getClassName('bpk-price__column-container')}
```

File: packages/bpk-component-price/src/BpkPrice.js [hunk 9]
```diff
@@ -63,9 +62,15 @@ const BpkPrice = (props: Props) => {
...
+      >
         <BpkText
           textStyle={isSmall ? TEXT_STYLES.heading4 : TEXT_STYLES.xxl}
+          className={getClassName(
+            'bpk-price__title',
+            !isAlighRight && 'bpk-price__spacing',
```
Comment (olliecurtis) [inline packages/bpk-component-price/src/BpkPrice.js:72]:
```suggestion
            !isAlignRight && 'bpk-price__spacing',
```

File: packages/bpk-component-price/src/BpkPrice.module.scss [hunk 10]
```diff
@@ -35,21 +35,27 @@
     text-decoration-line: line-through;
   }
 
-  &__titleColumn {
+  &__columnContainer {
```
Comment (olliecurtis) [inline packages/bpk-component-price/src/BpkPrice.module.scss:38]: Can this be, to follow BEM
```suggestion
  &__column-container {
```

**Review by olliecurtis — CHANGES_REQUESTED** [review]

**Review by olliecurtis — APPROVED** [review]

**Pull request merged**